packages/wallet/frontend/src/pages/transfer/request.tsx

```diff
@@ -314,20 +329,28 @@ type SelectAccountOption = SelectOption &
   }
 export const getServerSideProps: GetServerSideProps<{
   accounts: SelectAccountOption[]
+  incomingOutgoingBalances: Record<string, number>
 }> = async (ctx) => {
   const [accountsResponse] = await Promise.all([
     accountService.list(ctx.req.headers.cookie)
   ])
 
-  if (!accountsResponse.success) {
+  if (!accountsResponse.success || !accountsResponse.result) {
     return {
       notFound: true
     }
   }
 
-  if (!accountsResponse.result) {
-    return {
-      notFound: true
+  const incomingOutgoingBalances: Record<string, number> = {}
```

Comment:
Should we share incoming/outgoing balance state so that we don't calculate it in each page as we do here (for example)?